### PR TITLE
[DPE-3959] Update docs

### PR DESCRIPTION
**Pull request by zmraul**
Adds initial README information and sync with [discourse Karapace docs](https://discourse.charmhub.io/t/charmed-karapace-documentation/14178)

### CODE REVIEW

File: .github/workflows/sync_docs.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  workflow_call:
+
+jobs:
+  sync-docs:
+    name: Open PR with docs changes
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Open PR with docs changes
+        uses: deusebio/discourse-gatekeeper@main
```
Comment (deusebio) [inline .github/workflows/sync_docs.yaml:17]: we can now use the official

Comment (deusebio) [reply .github/workflows/sync_docs.yaml:17]: https://github.com/canonical/spark-k8s-bundle/blob/main/.github/workflows/sync_docs.yaml

File: .github/workflows/sync_docs.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  pull_request:
+
+jobs:
+  sync-docs:
+    name: Open PR with docs changes
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Open PR with docs changes
+        uses: canonical/discourse-gatekeeper@main
```
Comment (deusebio) [inline .github/workflows/sync_docs.yaml:18]: Please use the official one as here: https://github.com/canonical/spark-k8s-bundle/blob/main/.github/workflows/sync_docs.yaml

**Review by deusebio — CHANGES_REQUESTED** [review]
Just a small point, but to already be aligned with the latest and greatest

File: .github/workflows/sync_docs.yaml (hunk 2 above)
Comment (zmraul) [reply .github/workflows/sync_docs.yaml:18]: Already using that I think?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,6 +1,62 @@
...
+
+This documents explains the processes and practices recommended for contributing enhancements to this operator.
+
+- Generally, before developing enhancements to this charm, you should consider [opening an issue](https://github.com/canonical/karapace-operator/issues) explaining your problem with examples, and your desired use case.
+- If you would like to chat with us about your use-cases or proposed implementation, you can reach us at [Canonical Matrix public channel](https://matrix.to/#/#charmhub-data-platform:ubuntu.com) or [Discourse](https://discourse.charmhub.io/).
+- Familiarising yourself with the [Charmed Operator Framework](https://juju.is/docs/sdk) library will help you a lot when working on new features or bug fixes.
+- All enhancements require review before being merged. Code review typically examines
+  - code quality
+  - test coverage
+  - user experience for Juju administrators this charm.
```
Comment (Batalex) [inline CONTRIBUTING.md:13]: **typo**: There is a missing word, right?
```suggestion
  - user experience for Juju administrators of this charm.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,6 +1,62 @@
...
+
+## Requirements
+
+To build the charm locally, you will need to install [Charmcraft](https://juju.is/docs/sdk/install-charmcraft).
+
+To run the charm locally with Juju, it is recommended to use [LXD](https://linuxcontainers.org/lxd/introduction/) as your virtual machine manager. Instructions for running Juju on LXD can be found [here](https://juju.is/docs/olm/lxd).
+
+## Build and Deploy
+
+To build the charm in this repository, from the root of the dir you can run:
```
Comment (Batalex) [inline CONTRIBUTING.md:24]: **typo**: Duplicate with the line just below

**Review by Batalex — APPROVED** [review]
I don't want to block this PR for a typo, the content itself LGTM

File: .github/workflows/sync_docs.yaml (hunk 2 above)
Comment (deusebio) [reply .github/workflows/sync_docs.yaml:18]: yes, I believe I raised the comment before the last commit! Resolving!

**Review by deusebio — APPROVED** [review]

**Pull request merged**